streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -75,6 +75,9 @@
     // all internal topics auto-created by the topology builder and used in source / sink processors
     private final Set<String> internalTopicNames = new HashSet<>();
 
+    // map of internal topic names with topic configuration
+    private final Map<String, InternalTopicProperties> internalTopicNamesWithProperties = new HashMap<>();
```

Comment:
It seems like this might be able to just replace `internalTopicNames`. We only add to `internalTopicNames` in one place, where we also (maybe) add to this map. We can't (shouldn't) make the value `null`, but I noticed that `InternalTopicProperties` allows its parameter (`numberOfPartitions`) to be null, which seems like it should have the same effect as a null `internalTopicProperties`...

What do you think about requiring `iternalTopicProperties` to be non-null in `addInternalTopic`, although it might have a null number of partitions. Then, we can get rid of `internalTopicNames` and just use `internalTopicNamesWithProperties.keySet()`?

Reply:
Done.